src/main/java/bio/terra/stairway/Stairway.java

```diff
@@ -258,20 +279,37 @@ public Stairway(Stairway.Builder builder) {
             ? "stairway" + UUID.randomUUID().toString()
             : builder.stairwayName;
 
-    this.stairwayClusterName =
+    String stairwayClusterName =
         (builder.stairwayClusterName == null)
             ? "stairwaycluster" + UUID.randomUUID().toString()
             : builder.stairwayClusterName;
 
-    this.classLoader =
-        (builder.classLoader == null) ? this.getClass().getClassLoader() : builder.classLoader;
-
     this.flightFactory =
         (builder.flightFactory == null) ? new StairwayFlightFactory() : builder.flightFactory;
 
+    this.workQueueProjectId = builder.workQueueProjectId;
+    this.workQueueEnabled = (workQueueProjectId != null && builder.enableWorkQueue);
+    if (!workQueueEnabled) {
+      workQueueCreate = false;
+      this.workQueueTopicId = null;
+      this.workQueueSubscriptionId = null;
+    } else {
+      if (builder.workQueueTopicId == null) {
+        workQueueCreate = true;
+        this.workQueueTopicId = stairwayClusterName + "-workqueue";
+        this.workQueueSubscriptionId = stairwayClusterName + "-workqueue-sub";
+      } else {
+        workQueueCreate = false;
+        this.workQueueTopicId = builder.workQueueTopicId;
+        this.workQueueSubscriptionId = builder.workQueueSubscriptionId;
+        if (workQueueSubscriptionId == null) {
+          throw new StairwayExecutionException(
+              "subscription id must be specified if topic id is specified");
+        }
+      }
```

Comment:
I wanted to check if this reformulation made the code clearer in my mind. I'm not sure it does, although it's a bit more concise. Feel free to ignore.

```suggestion
      workQueueCreate = builder.workQueueTopicId == null;
      this.workQueueTopicId = workQueueCreate ? stairwayClusterName + "-workqueue" : builder.workQueueTopicId;
      this.workQueueSubscriptionId = workQueueCreate ? stairwayClusterName + "-workqueue-sub" : builder.workQueueSubscriptionId;
       if (workQueueSubscriptionId == null) {
          throw new StairwayExecutionException(
              "subscription id must be specified if topic id is specified");
       }
```

Reply:
The reason i did it the way i did it is to allow the TopicId and SubscriptionId to be final.